### PR TITLE
The smart_scraper_multi_graph method is too expensive

### PR DESCRIPTION
**Pull request by shenghongtw**
The smart_scraper_multi_graph method uses LLMs to answer questions both in the smart_scraper_graph and at the MergeAnswer node, which leads to higher costs. I created a new method called SmartScraperMultiParseMergeFirstGraph, which only uses LLMs at the MergeAnswer node. This reduces the cost by half for the same use case and achieve similar results, and the execution time is also shortened (as shown in the figure).
image
image
image

### CODE REVIEW

File: scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py
Comment (VinciGit00) [inline scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py:1]: Please instead of creating a new graph modify the original one

**Review by VinciGit00 — CHANGES_REQUESTED** [review]

File: scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py
Comment (shenghongtw) [reply scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py:1]: Modify the smart_scraper_multi_graph.py?

Comment (shenghongtw) [reply scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py:1]: But original method, called the [raptor](https://ragflow.io/blog/long-context-rag-raptor) method, is useful for deep extraction or in more complex situation.
My idea is to rename the original smart_scraper_multi_graph.py to smart_scraper_multi_raptor_graph.py, and then name my method smart_scraper_multi_graph.py.
What are your thoughts on this?
thx

Comment (VinciGit00) [reply scrapegraphai/graphs/smart_scraper_multi_parse_merge_first_graph.py:1]: Ok it looks nice

**Comment by VinciGit00** [comment]
please can you add in the examples of openai the configuration?

**Comment by VinciGit00** [comment]
and please can you call smart_scraper_multi_lite instead of that?

**Comment by shenghongtw** [comment]
> and please can you call smart_scraper_multi_lite instead of that?

you mean rename my method name to smart_scraper_multi_lite?

**Comment by shenghongtw** [comment]
> please can you add in the examples of openai the configuration?

ok

**Comment by VinciGit00** [comment]
Yes please rename it

**Comment by shenghongtw** [comment]
The original smart_scraper_multi_graph keeps its original name?

**Comment by VinciGit00** [comment]
> The original smart_scraper_multi_graph keeps its original name?

yes, just change your scraper name, in this way it will be more understandable at the first look

**Comment by VinciGit00** [comment]
and please add in examples/openai your script 
Screenshot 2024-10-17 alle 21 03 42

**Comment by VinciGit00** [comment]
Hi @shenghongtw, would you like to create more tests on the test folder please?

**Comment by shenghongtw** [comment]
> Hi @shenghongtw, would you like to create more tests on the test folder in another pull request please?

No problem.

**Comment by VinciGit00** [comment]
Hi @shenghongtw I will merge this pr now, when the tests will be ready I will be glad to merge it also them.
Thank you and have a nice day

**Pull request merged**